scopes/defender/jest/jest.main.runtime.ts

```diff
@@ -10,19 +11,30 @@ const jestM = require('jest');
 export const WORKER_NAME = 'jest';
 
 export class JestMain {
-  constructor(private jestWorker: HarmonyWorker<JestWorker>, private logger: Logger) {}
+  constructor(
+    private jestWorker: HarmonyWorker<JestWorker>,
+
+    /**
+     * Workspace extension.
+     */
+    private workspace: Workspace,
+    /**
+     * logger extension.
+     */
+    private logger: Logger
+  ) {}
 
   createTester(jestConfig: any, jestModule = jestM) {
-    return new JestTester(JestAspect.id, jestConfig, jestModule, this.jestWorker, this.logger);
+    return new JestTester(JestAspect.id, jestConfig, jestModule, this.jestWorker, this.logger, this.workspace);
   }
 
   static runtime = MainRuntime;
-  static dependencies = [WorkerAspect, LoggerAspect];
+  static dependencies = [WorkerAspect, LoggerAspect, WorkspaceAspect];
```

Comment:
I really don't like that jest is dependent on workspace. let's re-think a little bit about this solution

Reply:
@GiladShoham 
if I use the component from TesterContext I don't get the new content of the file, I must request it to check if the component filesystem is different.